### PR TITLE
Apply automated Sourcery refactorings to `plasmapy.plasma`

### PR DESCRIPTION
**Pull request by namurphy**
This PR uses the [Sourcery](https://sourcery.ai/) tool to automatically refactor `plasmapy.plasma` (along with pre-commit auto-fixes).

### CODE REVIEW

File: plasmapy/plasma/grids.py [hunk 1]
```diff
@@ -176,32 +176,29 @@ def require_quantities(self, req_quantities, replace_with_zeros=False):
             # Error check that grid contains E and B variables required
             if rq not in self.quantities:
 
-                # If missing, warn user and then replace with an array of zeros
-                if replace_with_zeros:
-                    warnings.warn(
-                        f"{rq} is not specified for the provided grid."
-                        "This quantity will be assumed to be zero.",
-                        RuntimeWarning,
+                if not replace_with_zeros:
```
Comment (namurphy) [inline plasmapy/plasma/grids.py:179]:
```suggestion
                # If missing, warn user and then replace with an array of zeros
                if not replace_with_zeros:
```
Add comment back in

File: plasmapy/plasma/tests/test_grids.py [hunk 2]
```diff
@@ -977,7 +977,7 @@ def debug_volume_avg_interpolator():
     analytic = interp_hax ** 4
 
     raw_hax = grid.ax0.to(u.mm).value
-    half = int(25 / 2)
+    half = 25 // 2
```
Comment (rocco8773) [inline plasmapy/plasma/tests/test_grids.py:980]:
```suggestion
    half = 12
```

File: plasmapy/plasma/sources/plasmablob.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 from plasmapy.formulary.collisions import coupling_parameter
 from plasmapy.formulary.dimensionless import quantum_theta
-from plasmapy.formulary.misc import _grab_charge
+from plasmapy.formulary.parameters import _grab_charge
```
Comment (rocco8773) [inline plasmapy/plasma/sources/plasmablob.py:11]:
```suggestion
from plasmapy.formulary.misc import _grab_charge
```

Comment (rocco8773) [inline plasmapy/plasma/sources/plasmablob.py:11]: Instead just import from `plasmapy.formulary`.

File: plasmapy/plasma/grids.py [hunk 4]
```diff
@@ -177,31 +177,29 @@ def require_quantities(self, req_quantities, replace_with_zeros=False):
...
+                    raise KeyError(
+                        f"{rq} is not specified for the provided "
+                        "grid but is required."
                     )
+                warnings.warn(
```
Comment (rocco8773) [inline plasmapy/plasma/grids.py:185]:
```suggestion

                warnings.warn(
```

File: plasmapy/plasma/grids.py [hunk 5]
```diff
@@ -1159,14 +1141,8 @@ def nearest_neighbor_interpolator(
...
         # Apply units to output arrays
-        output = []
-        for arg in range(len(args)):
-            output.append(vals[..., arg] * self._interp_units[arg])
-
-        if len(output) == 1:
-            return output[0]
-        else:
-            return tuple(output)
+        output = [vals[..., arg] * self._interp_units[arg] for arg in range(len(args))]
```
Comment (rocco8773) [inline plasmapy/plasma/grids.py:1144]:
```suggestion
        output = [vals[..., index] * self._interp_units[index] for index, _ in enumerate(args))]
```

File: changelog/1464.trivial.rst [hunk 6]
```diff
@@ -0,0 +1 @@
+ Applied automated refactorings from Sourcery to `plasmapy.plasma`.
```
Comment (rocco8773) [inline changelog/1464.trivial.rst]: Provide a link to Sourcery.

**Review by rocco8773 — APPROVED** [review]

**Pull request merged**